### PR TITLE
add dist algo

### PR DESCRIPTION
**Pull request by littletomatodonkey**
att

### CODE REVIEW

File: ppcls/loss/dist_loss.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def forward(self, z_s, z_t):
+        y_s = F.softmax(z_s, axis=-1)
+        y_t = F.softmax(z_t, axis=-1)
+        inter_loss = inter_class_relation(y_s, y_t)
+        intra_loss = intra_class_relation(y_s, y_t)
+        kd_loss = self.beta * inter_loss + self.gamma * intra_loss
+        return kd_loss
+
+
+# if __name__ == "__main__":
```
Comment (cuicheng01) [inline ppcls/loss/dist_loss.py]: 这个要不删掉？

File: ppcls/loss/dist_loss.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# copyright (c) 2021 PaddlePaddle Authors. All Rights Reserve.
```
Comment (cuicheng01) [inline ppcls/loss/dist_loss.py]: 2022

File: docs/zh_CN/advanced_tutorials/knowledge_distillation.md [hunk 3]
```diff
@@ -444,6 +445,73 @@ Loss:
     - CELoss:
         weight: 1.0
 ```
+
+
```
Comment (cuicheng01) [inline docs/zh_CN/advanced_tutorials/knowledge_distillation.md:450]: 加个索引？下同。

File: docs/zh_CN/advanced_tutorials/knowledge_distillation.md [hunk 4]
```diff
@@ -444,6 +445,73 @@ Loss:
...
+在ImageNet1k公开数据集上，效果如下所示。
+
+| 策略 | 骨干网络 | 配置文件 | Top-1 acc | 下载链接 |
+| --- | --- | --- | --- | --- |
+| baseline | ResNet18 | [ResNet18.yaml](../../../ppcls/configs/ImageNet/ResNet/ResNet18.yaml) | 70.8% | - |
+| DIST | ResNet18 | [resnet34_distill_resnet18_dist.yaml](../../../ppcls/configs/ImageNet/Distillation/resnet34_distill_resnet18_dist.yaml) | 71.99%(**+1.19%**) | - |
+
+
+##### 1.2.6.2 DKD 配置
```
Comment (cuicheng01) [inline docs/zh_CN/advanced_tutorials/knowledge_distillation.md]: DIST配置？

File: ppcls/configs/ImageNet/Distillation/resnet34_distill_resnet18_dist.yaml [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  Eval:
+    - CELoss:
+        weight: 1.0
+
+Optimizer:
+  name: Momentum
+  momentum: 0.9
+  weight_decay: 1e-4
+  lr:
+    name: MultiStepDecay
```
Comment (cuicheng01) [inline ppcls/configs/ImageNet/Distillation/resnet34_distill_resnet18_dist.yaml]: 这个和piecewise是一样的吧？要不换过来？

File: ppcls/loss/dist_loss.py (hunk 1 above)
Comment (littletomatodonkey) [reply ppcls/loss/dist_loss.py]: done

File: ppcls/loss/dist_loss.py (hunk 2 above)
Comment (littletomatodonkey) [reply ppcls/loss/dist_loss.py]: done

File: docs/zh_CN/advanced_tutorials/knowledge_distillation.md (hunk 3 above)
Comment (littletomatodonkey) [reply docs/zh_CN/advanced_tutorials/knowledge_distillation.md:450]: 1.2.6节已补充索引，多谢

File: docs/zh_CN/advanced_tutorials/knowledge_distillation.md (hunk 4 above)
Comment (littletomatodonkey) [reply docs/zh_CN/advanced_tutorials/knowledge_distillation.md]: fixed

File: ppcls/configs/ImageNet/Distillation/resnet34_distill_resnet18_dist.yaml (hunk 5 above)
Comment (littletomatodonkey) [reply ppcls/configs/ImageNet/Distillation/resnet34_distill_resnet18_dist.yaml]: 是的，已修改，多谢

**Review by cuicheng01 — APPROVED** [review]
